src/TestSessionEnvironment.php

```diff
@@ -163,7 +166,9 @@ public function startTestSession($state = null, $id = null)
         $json = json_encode($state, JSON_FORCE_OBJECT);
         $state = json_decode($json);
 
-        $this->applyState($state);
+        $state = $this->createDatabase($state);
```

Comment:
I'd guess you also need `mergeWithExistingState` in here

Reply:
> I'd guess you also need `mergeWithExistingState` in here
hmm I'm not sure about that. mergeWithExistingState is mainly useful for adding several properties from another object - it doens't do anything with the state object other than mutate it, which is what createDatabase does. The request ends just after this so it is **maybe** not a problem but perhaps we need to have a 'setState' to persist these changes on the class property.

Reply:
the only reason for that is backwards compatibility. No need for that anymore